test/TBD/dylib-version.swift

```diff
@@ -20,8 +20,8 @@
 
 // COMPAT: compatibility-version: 2
 
-// Same as above -- current version defaults to 1 and is not present in
-// emitted TBD file if it's 1.
-// COMPAT-NOT: current-version: 1
+// Same as above -- current version defaults to 0 and is not present in
```

Comment:
current version should default to 1, is that not the behavior you saw when removing the hardcoded value?

Reply:
I wasn't sure if 0 or 1 was the correct default. For the longest time I swore it was 1, but TAPI started producing 0 as the default. Maybe this is masking a bug in TAPI, but I'd rather TAPI and Swift be in sync than Swift hard-coding a value.

Reply:
1 is the correct default, we talked offline and it seems TAPI is doing the wrong thing. I'll adjust this test case back to what it was, then XFAIL it based on the TAPI radar. But I'd like to preserve the consistency between TAPI and Swift, if that makes sense.

Reply:
There are two separate defaults at play here:

The first one is the linker default value for these versions. The linker assumes a version number of 0 for current version and compatibility version if they are not specified on the command line.

The second one is the default value that is used by the TBD file encoding. The TBD file is optimized to minimize output and therefore has different default values that allow it to omit entries. 

In this case Swift needs to behave like the linker and assume a default version of 0 if no command line option is provided and pass on that value to the TBD writer.

Reply:
So is this the right fix, or should I set it to the "real" default of `{0, 0, 0}`?

Reply:
yes, sorry for causing confusion. If omitting the version on command line, the tbd `current version` should indeed be written 0 in the tbd file. if the `current version` is set to 1 on the command line, tbd `current version` will just be omitted from the file.